js/src/setup-mc/setup-stepper/setup-free-listings/index.js

```diff
@@ -115,6 +116,9 @@ const SetupFreeListings = ( props ) => {
 					shipping_time: settings.shipping_time,
 					tax_rate: settings.tax_rate,
 					shipping_country_rates: dataShippingRates,
+					offer_free_shipping: getOfferFreeShippingInitialValue(
+						dataShippingRates
+					),
```

Comment:
Hmm right, I think that is the only caveat we have to live with, because we don't store or retrieve the "yes"/"no" value from the API, instead we derive the "yes"/"no" value from the existence of `shippingRate.options.free_shipping_threshold`. So, when users make a page reload, we retrieve the shipping rates, and the initial value would be "no" as shown in your video.

If we were to make the initial value `undefined`, then it might turn out to be "buggy" too. Imagine if users selected the target audience countries, entered the shipping rates, select "no", and then reload the page. The options won't be checked (because we make initial value `undefined`).

So I guess in the end, we have to accept this "page reload" caveat. 😅

Reply:
> I think that is the only caveat we have to live with, because we don't store or retrieve the "yes"/"no" value from the API

Using `localStorage` or `sessionStorage` might be an acceptable approach for this case.

> As a user, I expect when I fill a form, if an important field changes, the related important fields must be reset, so that I can be prompted to rethink and confirm that those field values are correct by filling them again.

⬆️  It's quoted from the discussion in https://github.com/woocommerce/google-listings-and-ads/pull/1268#discussion_r813568572.

It's not the same context and operation process here but assumed the concept is similar to it. For the same operating steps, it should be more "natural" and less surprising to see a selected radio option being reset than an untouched radio option being automatically selected after refreshing the page.

---

Given that this is a rare use case, allowing this caveat to exist is a trade-off. After all, even using localStorage/sessionStorage does not cover all possibilities. I think this concern is downgraded to a 💅  or 📜  now.